src/test/java/htsjdk/variant/variantcontext/VariantContextBuilderTest.java

```diff
@@ -47,19 +55,207 @@ public void testAttributeResettingWorks(final boolean leaveModifyableAsIs) {
+                .attribute("AC", 1)
+                .attribute("AN", 2)
+                .make();
 
-        final VariantContext result1 = root1.attribute("AC", 1).make();
+        final VariantContextBuilder root2 = new VariantContextBuilder(result1);
 
         //this is a red-herring and should not change anything.
         final VariantContext ignored = root1.attribute("AC", 2).make();
 
+        final VariantContext result2 = root2.make();
+
+        Assert.assertEquals(result1.getAttribute("AC"), result2.getAttribute("AC"));
+        Assert.assertEquals(result1.getAttribute("AN"), result2.getAttribute("AN"));
+
+        final HashMap<String, Object> map = new HashMap<>();
+        map.put("AC", 2);
+        map.put("AN", 4);
+        root2.attributes(map);
+
+        final VariantContext result3 = root2.make();
+
+        Assert.assertEquals(result3.getAttribute("AC"), 2);
+        Assert.assertEquals(result3.getAttribute("AN"), 4);
+    }
+
+    enum VCBuilderScheme {
+        NEW_ON_BUILDER {
+            @Override
+            VariantContextBuilder getOtherBuilder(final VariantContextBuilder builder, final VariantContext vc) {
+                return new VariantContextBuilder(builder);
+            }
+        },
+        NEW_ON_VC {
+            @Override
+            VariantContextBuilder getOtherBuilder(final VariantContextBuilder builder, final VariantContext vc) {
+                return new VariantContextBuilder(vc);
+            }
+        },
+        SAME {
+            @Override
+            VariantContextBuilder getOtherBuilder(final VariantContextBuilder builder, final VariantContext vc) {
+                return builder;
+            }
+        };
+
+        abstract VariantContextBuilder getOtherBuilder(final VariantContextBuilder builder, VariantContext vc);
+    }
+
+    @DataProvider
+    public Object[][] BuilderSchemes() {
+        return new Object[][]{
+                {VCBuilderScheme.NEW_ON_BUILDER},
+                {VCBuilderScheme.NEW_ON_VC},
+                {VCBuilderScheme.SAME}};
+    }
+
+    @Test(dataProvider = "BuilderSchemes")
+    public void testAttributeResettingWorks(final VCBuilderScheme builderScheme) {
+        final VariantContextBuilder root1 = new VariantContextBuilder(snpSource, snpChr, snpLocStart, snpLocStop, Arrays.asList(Aref, C));
+
+        final VariantContext result1 = root1.attribute("AC", 1).make();
+        final VariantContextBuilder root2 = builderScheme.getOtherBuilder(root1, result1)
+                .source(snpSource).chr(snpChr).start(snpLocStart).stop(snpLocStop).alleles(Arrays.asList(Aref, C));
+
+        //this is a red-herring and should not change anything.
+        final VariantContext ignored = root1.attribute("AC", 2).make();
         final VariantContext result2 = root2.attribute("AC", 1).make();
 
-       Assert.assertEquals(result1.getAttribute("AC"), result2.getAttribute("AC"));
+        Assert.assertEquals(result1.getAttribute("AC"), result2.getAttribute("AC"));
```

Comment:
One of the potential issues around the builder is that there can be subtle bugs that relate to the order of instantiation vs alteration.

```
with this initialization:
builder1.attribute("AC", 1)
builder2 = new Builder(builder1)

these two following operations could be different:

case1: 
result1 = builder1.make()
builder2.attribute("AC", 2)
result2= builder2.make()

case2:
builder2.attribute("AC", 2)
result1 = builder1.make()
result2= builder2.make()
```

That could be the case if the builder has an internal state that is accidentally shared/modified but the `make()` operation performs a copy. In both cases I would expect `result1` to have ac of 1 but it's possible that case 1 would be fine but case 2 would be broken.  I don't see a test for that type of difference here.  I think in practice the only place that make() copies anything is with the alleles, so this may be unnecessary / overkill to tests.

Reply:
case 1 is tested already, I believe, but I'll add it explicitly, also case 2 works, but I added a test.